app/grandchallenge/algorithms/forms.py

```diff
@@ -13,18 +19,109 @@
     AlgorithmPermissionRequest,
     Job,
 )
+from grandchallenge.components.models import InterfaceKind
 from grandchallenge.core.forms import (
     PermissionRequestUpdateForm,
     SaveFormInitMixin,
     WorkstationUserFilterMixin,
 )
 from grandchallenge.core.validators import ExtensionValidator
-from grandchallenge.core.widgets import MarkdownEditorWidget
+from grandchallenge.core.widgets import JSONEditorWidget, MarkdownEditorWidget
 from grandchallenge.groups.forms import UserGroupForm
 from grandchallenge.jqfileupload.widgets import uploader
 from grandchallenge.jqfileupload.widgets.uploader import UploadedAjaxFileList
+from grandchallenge.reader_studies.models import ANSWER_TYPE_SCHEMA
 from grandchallenge.subdomains.utils import reverse_lazy
 
+image_upload_text = (
+    "The total size of all files uploaded in a single session "
+    "cannot exceed 10 GB.<br>"
+    "The following file formats are supported: "
+    ".mha, .mhd, .raw, .zraw, .dcm, .nii, .nii.gz, "
+    ".tiff, .png, .jpeg and .jpg.<br>"
+    "The following file formats can be uploaded and will be converted to "
+    "tif: Aperio(.svs), Hamamatsu(.vms, .vmu, .ndpi), Leica(.scn), MIRAX"
+    "(.mrxs) and Ventana(.bif)."
+)
+
+file_upload_text = (
+    "The total size of all files uploaded in a single session "
+    "cannot exceed 10 GB.<br>"
+    "The following file formats are supported: "
+)
+
+
+class InterfaceFormField:
+    def __init__(
+        self, kind: InterfaceKind.InterfaceKindChoices, initial=None, user=None
+    ):
+        field_type = field_for_interface(kind)
+
+        # bool can't be required
+        kwargs = {
+            "required": (kind != InterfaceKind.InterfaceKindChoices.BOOL)
+        }
+        if initial:
+            kwargs["initial"] = initial
+        if kind in InterfaceKind.interface_type_annotation():
+            kwargs["widget"] = JSONEditorWidget(
+                schema=ANSWER_TYPE_SCHEMA["definitions"][kind]
+            )
+        if kind in InterfaceKind.interface_type_file():
+            kwargs["widget"] = uploader.AjaxUploadWidget(
+                multifile=False, auto_commit=False
+            )
+            kwargs["help_text"] = f"{file_upload_text} .{kind.lower()}"
+            kwargs["validators"] = [
+                ExtensionValidator(allowed_extensions=(f".{kind.lower()}",))
+            ]
+        if kind in InterfaceKind.interface_type_image():
+            kwargs["widget"] = uploader.AjaxUploadWidget(
+                multifile=True, auto_commit=False
+            )
+            kwargs["help_text"] = image_upload_text
+        self._field = field_type(**kwargs)
+        if user:
+            self._field.widget.user = user
+
+    @property
+    def field(self):
+        return self._field
+
+
+def field_for_interface(i: InterfaceKind.InterfaceKindChoices):
+    if i == InterfaceKind.InterfaceKindChoices.BOOL:
```

Comment:
`InterfaceKind.InterfaceKindChoices.BOOL` is repeated 3 times here, style wise, wouldn't a dictionary make more sense?

Reply:
I agree. I'll change it.